deepchecks/vision/vision_data/vision_data.py

```diff
@@ -0,0 +1,239 @@
+
+    @property
+    def has_embeddings(self) -> bool:
+        """Return True if the dynamic loader contains embeddings."""
+        return self._has_embeddings
+
+    @property
+    def has_additional_data(self) -> bool:
+        """Return True if the dynamic loader contains additional_data."""
+        return self._has_embeddings
+
+    @property
+    def has_image_identifiers(self) -> bool:
+        """Return True if the dynamic loader contains image identifiers."""
+        return self._has_image_identifiers
+
+    @property
+    def task_type(self) -> TaskType:
+        """Return True if the dynamic loader contains labels."""
+        return self._task_type
+
+    @property
+    def dynamic_loader(self):
+        """Return the dynamic loader used be the vision data."""
+        return self._dynamic_loader
+
+    @property
+    def number_of_images_cached(self) -> int:
+        """Return True if the number of images processed and whose statistics were cached."""
+        return self._num_images_seen
+
+    @property
+    def observed_classes(self) -> t.Dict[t.Union[int, str], int]:
+        """Return a dictionary of observed classes and their statistics from internal cache."""
+        return self._observed_classes
+
+    def label_id_to_name(self, class_id: int) -> str:
+        """Return the name of the class with the given id."""
+        class_id = int(class_id)
+        if self._label_map is None:
+            return str(class_id)
+        elif class_id not in self._label_map:
+            get_logger().warning('Class id %s is not in the label map. Add it to map '
+                                 'in order to show the class name instead of id', class_id)
+            return str(class_id)
+        else:
+            return self._label_map[class_id]
+
+    def copy(self, reshuffle_dynamic_loader: bool = False, dynamic_loader=None) -> VD:
+        """Create new copy of the vision data object with clean cache.
+
+        Parameters
+        ----------
+        reshuffle_dynamic_loader: bool, default=True
+            If True and the dynamic loader is of known type that can be shuffled, it will be shuffled.
+        dynamic_loader:
+            If not None, the dynamic loader of the new object will be set to this value.
+        model:
+            If not None, the model of the new object will be set to this value.
+        Returns
+        -------
+        VisionData
+            A copy of the vision data object with clean cache.
+        """
+        cls = type(self)
+        dynamic_loader = dynamic_loader if dynamic_loader is not None else self._dynamic_loader
+        return cls(dynamic_loader=dynamic_loader, task_type=self._task_type.value, label_map=self._label_map,
+                   dataset_name=self.name, reshuffle_dynamic_loader=reshuffle_dynamic_loader)
+
+    def __iter__(self):
+        """Return an iterator over the dynamic loader."""
+        return iter(self._dynamic_loader)
+
+    def __len__(self):
```

Comment:
Just to make sure, reminding here to write a test on a non iterable loader.

Reply:
Of course